ExampleMod/Tiles/ExampleCommandSign.cs

```diff
@@ -0,0 +1,108 @@
+
+namespace ExampleMod.Tiles
+{
+	public class ExampleCommandSign : ModTile
+	{
+		public ExampleCommandSign()
+		{
+		}
+
+		public override void SetDefaults()
+		{
+			// Credits to Dark;Light for finding this flag
+			// The Main.tileSign flag will do the following:
+			//  *Automatically manages the sign for the specified tile
+			//   -Adds a right-click to the tile to bring up an edit sign window
+			//   -Allows editing of the sign text
+			//   -Saves and loads sign data to world file
+			Main.tileSign[Type] = true;
+			Main.tileFrameImportant[Type] = true;
+			Main.tileLavaDeath[Type] = true;
+
+			// Allow hanging from ceilings
+			TileObjectData.newTile.CopyFrom(TileObjectData.Style2x2);
+			TileObjectData.newAlternate.CopyFrom(TileObjectData.newTile);
+			TileObjectData.newAlternate.StyleHorizontal = true;
+			TileObjectData.newAlternate.Origin = new Point16(0, 0);
+			TileObjectData.newAlternate.AnchorTop = new AnchorData(AnchorType.SolidTile | AnchorType.SolidSide, 2, 0);
+			TileObjectData.newAlternate.AnchorBottom = AnchorData.Empty;
+			TileObjectData.addAlternate(1);
+
+			// Allow attaching to a solid object that is to the left of the sign
+			TileObjectData.newAlternate.CopyFrom(TileObjectData.newTile);
+			TileObjectData.newAlternate.StyleHorizontal = true;
+			TileObjectData.newAlternate.Origin = new Point16(0, 0);
+			TileObjectData.newAlternate.AnchorLeft = new AnchorData(AnchorType.SolidTile | AnchorType.SolidSide, 2, 0);
+			TileObjectData.newAlternate.AnchorBottom = AnchorData.Empty;
+			TileObjectData.addAlternate(2);
+
+			// Allow attaching to a solid object that is to the right of the sign
+			TileObjectData.newAlternate.CopyFrom(TileObjectData.newTile);
+			TileObjectData.newAlternate.StyleHorizontal = true;
+			TileObjectData.newAlternate.Origin = new Point16(0, 0);
+			TileObjectData.newAlternate.AnchorRight = new AnchorData(AnchorType.SolidTile | AnchorType.SolidSide, 2, 0);
+			TileObjectData.newAlternate.AnchorBottom = AnchorData.Empty;
+			TileObjectData.addAlternate(3);
+
+			// Allow attaching to a wall behind the sign
+			TileObjectData.newAlternate.CopyFrom(TileObjectData.newTile);
+			TileObjectData.newAlternate.StyleHorizontal = true;
+			TileObjectData.newAlternate.Origin = new Point16(0, 0);
+			TileObjectData.newAlternate.AnchorWall = true;
+			TileObjectData.newAlternate.AnchorBottom = AnchorData.Empty;
+			TileObjectData.addAlternate(4);
+			TileObjectData.addTile(Type);
+			ModTranslation name = CreateMapEntryName();
+			name.SetDefault("Example Command Sign");
+			AddMapEntry(new Color(200, 200, 200), name);
+			dustType = mod.DustType("Sparkle");
+			disableSmartCursor = true;
+			adjTiles = new int[] { Type };
+		}
+
+		public override bool HasSmartInteract()
+		{
+			return true;
+		}
+
+		public override void RightClick(int i, int j)
+		{
+			// Uses the text from the sign to run a command
+			Main.ExecuteCommand(Main.sign[Sign.ReadSign(i, j, true)].text, new ExampleCommandCaller());
```

Comment:
Just making sure, this will work fine if the text passed isn't a valid command?

Reply:
It works fine; there is a try/catch block around the all the logic that will catch invalid commands.